src/subscription_manager/identity.py

```diff
@@ -37,11 +37,11 @@ class ConsumerIdentity:
 
     @classmethod
     def keypath(cls):
-        return Path.join(cls.PATH, cls.KEY)
+        return str(Path.join(cls.PATH, cls.KEY))
 
     @classmethod
     def certpath(cls):
-        return Path.join(cls.PATH, cls.CERT)
+        return str(Path.join(cls.PATH, cls.CERT))
```

Comment:
Yes, dnf configuration returns unicode and it makes subscription managed code fail somewhere down the path.